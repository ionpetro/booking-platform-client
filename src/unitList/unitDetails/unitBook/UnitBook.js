import Button from '../../../components/Button/Button.vue';

export default {
  components: {
    'v-button': Button
  },
  props: {
    unit: {
      required: true
    }
  },
  computed: {
    hasDiscount() {
      return !(this.unit.price === this.unit.discountedPrice);
    }
  }
};
