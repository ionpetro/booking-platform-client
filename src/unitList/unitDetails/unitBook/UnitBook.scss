.book {
  &__container {
    display: flex;
    align-items: center;
    justify-content: space-between;
    // push content to left so the content is aligned
    margin-right: 24px;

    &__price {
      padding: 24px 0 26px 24px;
      margin: 0;
      &__amount {
        font-weight: 700;
      }

      &__discount {
        color: $red-medium;
        text-decoration: line-through;
        padding-right: 2px;
      }

      @media #{$mobile} {
        // align price with container
        padding: 26px 0 26px 16px;
      }
    }

    @media #{$mobile} {
      // adapt content to align with container
      margin-right: 16px;
    }
  }
}
