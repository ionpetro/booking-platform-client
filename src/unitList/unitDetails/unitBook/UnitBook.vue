<template>
  <div class="book">
    <div class="book__container">
      <div class="book__container__price">
        <span class="body--small book__container__price__discount" v-if="hasDiscount">{{
          unit.price
        }}</span>
        <span class="body--large book__container__price__amount"
          >{{ unit.discountedPrice }}
          <img src="@/assets/icons/Bitcoin.svg" alt="Bitcoin icon" />
        </span>
      </div>
      <div class="book__form">
        <UnitBookDate @onSelectedYear="yearSelected" :unit="unit"></UnitBookDate>
        <v-button @click.native="bookUnit" :disabled="!selectedYear" class="btn" variant="secondary"
          >Book</v-button
        >
      </div>
    </div>
  </div>
</template>

<script src="./UnitBook.js"></script>

<style lang="scss" src="./UnitBook.scss" scoped></style>
