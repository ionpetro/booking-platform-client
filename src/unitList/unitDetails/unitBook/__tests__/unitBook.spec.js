import {mount} from '@vue/test-utils';
import UnitBook from '../UnitBook.vue';
import unitMock from '../../__tests__/mocks/unit.mock';

describe('UnitBook', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(UnitBook, {
      propsData: {unit: unitMock}
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  describe('when loaded', () => {
    it('renders data correctly', async () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
