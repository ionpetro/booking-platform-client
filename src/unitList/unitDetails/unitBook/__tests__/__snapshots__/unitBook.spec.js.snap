// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitBook when loaded renders data correctly 1`] = `
<div
  class="book"
>
  <div
    class="book__container"
  >
    <div
      class="book__container__price"
    >
      <span
        class="body--small book__container__price__discount"
      >
        1500
      </span>
       
      <span
        class="body--large book__container__price__amount"
      >
        1100
        
        <img
          alt="Bitcoin icon"
          src="@/assets/icons/Bitcoin.svg"
        />
      </span>
    </div>
     
    <button
      class="btn button secondary"
    >
      Book
    </button>
  </div>
</div>
`;
